src/business/form.ts

```diff
@@ -1,44 +1,56 @@
-export class Form {
-
-    private values: { [key: string]: any } = {};
+export interface FieldState {
+    value: any;
+    validationError?: string;
+    pristine: boolean;
+}
 
-    private validationErrors: { [fieldName: string]: any } = {};
+export class Form {
 
+    private fieldStates: { [key: string]: FieldState } = {};
     private fieldListeners: { [fieldName: string]: Array<(value: any) => void> } = {};
 
-    public getFieldValue(fieldName: string): any {
-        return this.values[fieldName] || '';
+    public getFieldState(fieldName: string): FieldState | undefined {
+        return this.fieldStates[fieldName];
     }
 
-    public setFieldValues(values: { [fieldName: string]: any }) {
+    public setFieldValues(values: { [fieldName: string]: any }, pristineValues = false) {
```

Comment:
I'm not digging the magic third boolean, why not introduce something like a `resetFieldValues` method